pkg/skaffold/kubernetes/debugging/container_manager.go

```diff
@@ -64,51 +55,41 @@ func (d *ContainerManager) Start(ctx context.Context) error {
 		// debug mode probably not enabled
 		return nil
 	}
-	stopWatchers, err := aggregatePodWatcher(d.namespaces, d.aggregate)
+
+	d.podWatcher.Register(d.events)
```

Comment:
+1, but it might be tricky because we actually add filters (tags) to the podSelector at runtime. maybe the podWatcher could automatically pick these up but I'm not sure

Reply:
I'll take a look at that in step 2.